src/app/features/charts/components/score-barchart/score-barchart.component.ts

```diff
@@ -71,19 +71,32 @@ export class ScoreBarChartComponent implements AfterViewInit, OnChanges {
     this.createChart();
   }
 
+  setScoreIndex(bins: number[][]) {
+    bins.forEach((item, index: number) => {
+      if (this._score === null)
+        return;
+      if (this._score >= item[0] && this._score < item[1]) {
+        this.scoreIndex = index;
+      }
+      if (index === bins.length - 1) {
+        // check border case where score is equal to the last bin upper bound
+        if (this._score === item[1])
+          this.scoreIndex = index;
+      }
+    });
+  }
+
   initData() {
+    if (!this.data)
+      return;
+    
     if (!this.data)
       return;
```

Comment:
Could one of these statements be removed?

Reply:
Removed.  Thanks!